jsapp/js/components/activity/activity.constants.ts

```diff
@@ -38,216 +39,169 @@ export enum AuditActions {
   },
-  'replace-form': {
-    order: 6,
-    name: AuditActions['replace-form'],
-    label: t('upload new form'),
-    message: t('##username## uploaded a new form'),
-  },
-  'share-data-publicly': {
-    order: 15,
-    name: AuditActions['share-data-publicly'],
-    label: t('share data publicly'),
-    message: t('##username## shared data publicly'),
-  },
-  'share-form-publicly': {
-    order: 14,
-    name: AuditActions['share-form-publicly'],
-    label: t('make project public'),
-    message: t('##username## made the project publicly accessible'),
-  },
-  transfer: {
-    order: 20,
-    name: AuditActions['transfer'],
-    label: t('transfer project ownership'),
-    message: t('##username## transferred project ownership to ##username2##'),
-  },
-  unarchive: {
-    order: 5,
-    name: AuditActions['unarchive'],
-    label: t('unarchive project'),
-    message: t('##username## unarchived project'),
-  },
-  'unshare-data-publicly': {
-    order: 18,
-    name: AuditActions['unshare-data-publicly'],
-    label: t('disable sharing data publicly'),
-    message: t('##username## disabled sharing data publicly'),
-  },
-  'unshare-form-publicly': {
-    order: 17,
-    name: AuditActions['unshare-form-publicly'],
-    label: t('disable making project public'),
-    message: t('##username## disabled making project publicly accessible'),
-  },
-  'update-content': {
-    order: 7,
-    name: AuditActions['update-content'],
-    label: t('edit form'),
-    message: t('##username## edited the form in the form builder'),
+  {
+    name: AuditActions['modify-service'],
+    label: t('modify a REST service'),
+    message: t('##username## modified a REST service'),
   },
-  'update-name': {
-    order: 0,
-    name: AuditActions['update-name'],
-    label: t('change name'),
-    message: t('##username## changed project name'),
+  {
+    name: AuditActions['delete-service'],
+    label: t('delete a REST service'),
+    message: t('##username## deleted a REST service'),
   },
-  'update-settings': {
-    order: 1,
-    name: AuditActions['update-settings'],
-    label: t('update settings'),
-    message: t('##username## updated project settings'),
+  { name: AuditActions['add-submission'], label: t('add submission'), message: t('##username## added a submission') },
+  {
+    name: AuditActions['modify-submission'],
+    label: t('edit submission'),
+    message: t('##username## edited a submission'),
   },
-  'update-qa': {
-    order: 8,
-    name: AuditActions['update-qa'],
-    label: t('modify qualitative analysis questions'),
-    message: t('##username## modified qualitative analysis questions'),
+  {
+    name: AuditActions['delete-submission'],
+    label: t('delete a submission'),
+    message: t('##username## deleted a submission'),
   },
+]
+
+type AuditActionTypes = {
+  [P in AuditActions]: AuditActionDefinition
 }
 
+export const AUDIT_ACTION_TYPES: AuditActionTypes = _AUDIT_ACTION_TYPES.reduce((acc, action, index) => {
```

Comment:
@Akuukis @pauloamorimbr what do you think about this? :)

Reply:
Looks good! Although not nice due linter :sweat: 

Why the `  if (action.name) {`?

Reply:
> Looks good! Although not nice due linter 😓
> 
> Why the ` if (action.name) {`?

Yeah linter made this a bit 🤢. I thin the `if` is a leftover from a WIP iteration of this. It is not needed now, good catch! Removed it

Reply:
@magicznyleszek also see https://biomejs.dev/formatter/#ignore-code, I think this place justifies ignoring formatter.